Sources/TokamakCore/Fiber/LayoutComputer.swift

```diff
@@ -17,9 +17,35 @@
 
 import Foundation
 
+/// The currently computed children.
+struct LayoutContext {
+  var children: [Child]
+
+  struct Child {
+    let index: Int
+    let dimensions: ViewDimensions
+  }
+}
+
 /// A type that is able to propose sizes for its children.
-protocol LayoutComputer {
+///
+/// The order of calls is guaranteed to be:
+/// 1. `proposeSize(for: child1, at: 0, in: context)`
+/// 2. `proposeSize(for: child2, at: 1, in: context)`
+/// 3. `position(child1)`
+/// 4. `position(child2)`
+///
+/// The `context` will contain all of the previously computed children from `proposeSize` calls.
+///
+/// The same `LayoutComputer` instance will be used for any given view during a single layout pass.
+///
+/// Sizes from `proposeSize` will be clamped, so it is safe to return negative numbers.
+protocol LayoutComputer: AnyObject {
   /// Will be called every time a child is evaluated.
   /// The calls will always be in order, and no more than one call will be made per child.
-  func proposeSize<V: View>(for child: V, at index: Int) -> CGSize
+  func proposeSize<V: View>(for child: V, at index: Int, in context: LayoutContext) -> CGSize
+  /// The child responds with their size and we place them relative to our origin.
+  func position(_ child: LayoutContext.Child, in context: LayoutContext) -> CGPoint
+  /// Request a size for ourself from our parent.
+  func requestSize(in context: LayoutContext) -> CGSize
```

Comment:
nit: I personally prefer a bit more spacing for easier reading
```suggestion

  /// The child responds with their size and we place them relative to our origin.
  func position(_ child: LayoutContext.Child, in context: LayoutContext) -> CGPoint

  /// Request a size for ourself from our parent.
  func requestSize(in context: LayoutContext) -> CGSize
```